### ISSUE TITLE
V3 `mergePrismicPreviewData` with null values

### ISSUE THREAD

**Issue by jaredloson** [angeloashmore/gatsby-source-prismic#139]
**Version 3.0.0-beta.10**

I'm getting an error when using `mergePrismicPreviewData` when it encounters null values: `TypeError: Cannot read property 'id' of null`

Seems to be related to this line: https://github.com/angeloashmore/gatsby-source-prismic/blob/v3-beta/src/browser/helpers.js#L217

A potential fix could be to change the `if` condition like so:

```
// typeof null === "object", so check that node is not falsy first
if (node && typeof node === 'object' && node.id === previewId) {
  ...
}
```

**Comment by chapster11** [comment]
Any fix on this issue ? I'm also getting this error. 
@jaredloson did you happen to find a work around or a fix yourself ?

**Comment by jaredloson** [comment]
@chapster11, or anyone else who may find this useful: in lieu of a fix, I've been using my own wrapper function on `mergePrismicPreviewData` in which I replace `null` values with `undefined` values. Seems to be working for me, but could be unintended consequences.

```
import { mergePrismicPreviewData } from "gatsby-source-prismic";
import traverse from "traverse";

function makeNullsUndefined(node) {
  if (node === null) {
    this.update(undefined);
  }
}

const mergePreviewData = staticData => {
  return mergePrismicPreviewData({
    staticData: traverse(staticData).map(makeNullsUndefined),
    previewData: traverse(window. __PRISMIC_PREVIEW_DATA__).map(makeNullsUndefined),
  });
};

export default mergePreviewData;
```

**Comment by chapster11** [comment]
@jaredloson thanks for this. Your node check in the if statement did work but the merge is not working of PreviewData and staticData. Is this working for you after you made your change ?

**Comment by jaredloson** [comment]
@chapster11 the merge does work for me after mapping nulls to undefined.

**Comment by angeloashmore** [comment]
Hi @jaredloson, thanks for the diagnosis. I'll publish a fix shortly with your changes.

**Comment by chapster11** [comment]
Thanks @angeloashmore that would be amazing.

**Comment by chapster11** [comment]
@angeloashmore don't mean to rush anyone here but wondering on an ETA on this fix. 
Obviously not a show stopper like the bug above but also wondering if the Deprecation warnings will be suppressed in this next build ? 

warn Deprecation warning - adding inferred resolver for field `........ In Gatsby v3, only fields with an explicit directive/extension will get a resolver.`

Not so bad when you have only one page but when you have over 50 it gets pretty busy in the console. 

Thanks for all the hard work on getting V3 to us all.

**Comment by angeloashmore** [comment]
Fixed in `v3.0.0-beta.14`. Thanks for being patient, I haven't had time away from work to focus on this.
